generators/pkg/generator/generate.go

```diff
@@ -25,3 +45,94 @@ func GenerateTask(spec GitHubSpec) v1beta1.Task {
 	)
 	return task
 }
+
+func addTask(url string) (v1beta1.Task, error) {
+	task := v1beta1.Task{}
+	response, err := http.Get(url)
+	if err != nil {
+		return task, fmt.Errorf("fail to get git-clone task from %s: %w", url, err)
+	}
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return task, fmt.Errorf("fail to read reponse body: %v", err)
+	}
+
+	if err := yaml.Unmarshal(body, &task); err != nil {
+		return task, fmt.Errorf("unable to unmarshal git-clone task: %w", err)
+	}
+	task.Labels = map[string]string{"generator.tekton.dev": task.Name}
+	return task, nil
+}
+
+func GeneratePipeline(github GitHub) (v1beta1.Pipeline, []v1beta1.Task, error) {
+	gitCloneTask, err := addTask(GitCloneURL)
+	if err != nil {
+		return v1beta1.Pipeline{}, []v1beta1.Task{}, fmt.Errorf("fail to add git-clone task: %w", err)
+	}
+	buildTask := GenerateTask(github)
+
+	ws := "source"
+	name := "build-from-git-repo"
+	tasksName := []string{"fetch-git-repo", "build-from-repo"}
+
+	pipeline := v1beta1.Pipeline{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+			Kind:       "Pipeline",
+		},
```

Comment:
@dibyom do you know if there's a way to generate this? This feels like the type of thing that k8s should know how to infer from the type.

Reply:
Hmm..yeah I don't think we need to explicitly specify the Type meta fields (i.e. Kind and APIVersion). Though I'm not sure where that gets added automatically